### ISSUE TITLE
Game not started on iOS and mac safari when targeting to openfl-html5 backend

### ISSUE THREAD

**Issue by fysx** [HaxeFlixel/flixel#1462]
when I target any haxeflixel project to openfl-html5 backend? game not loading on iOS and Mac Safari.
Error in console:
TypeError: undefined is not an object (evaluating 'a.apply') in sound.js.min:17:894

Project is clean (Hello world), all haxe libraries are up to date. Any projects on simple openFl works well.
##

**Comment by fysx** [comment]
if i removed following lines:

	urls.push("assets/sounds/beep.mp3");
	types.push("MUSIC");
	urls.push("assets/sounds/flixel.mp3");
	types.push("MUSIC");
	urls.push("assets/sounds/beep.ogg");
	types.push("SOUND");
	urls.push("assets/sounds/flixel.ogg");
	types.push("SOUND");

 in compiled js file, evereything works well

**Comment by fysx** [comment]
I converted beep.mp3 and flixel.mp3. It helps.

**Comment by Masadow** [comment]
This issue should not be closed

**Comment by AGulev** [comment]
This issue should not be closed

**Comment by sruloart** [comment]
Safari can't play .ogg files, so you better use .wav files, .m4a files, or disregard the legal implications and use .mp3 files. 

You should also make sure that .ogg files aren't being included with the build `unless="IE | Safari"` (not sure if there's a Safari flag actually, https://github.com/HaxeFlixel/flixel/blob/3.3.11/include.xml) or just delete the default files directly from your assets folder.

**Comment by MSGhero** [comment]
@sruloart You can't reasonably haxeflag for each browser, it's not like you can control which version gets loaded per user unless it's running on your own site. It looks like there are some plugins for Safari, and you might link the player to one of those, but I have no idea if they work.

**Comment by sruloart** [comment]
@MSGhero I'm not great (or modestly good) with words, I can promise you there's no "IE" flag either :D I just meant, "don't include .ogg files if you intend to use IE or Safari" and forgot to add a happy face to express cynicism :) 

It is possible to play other audio formats on Safari (as well as on IE btw), it's just not possible to include .ogg files without a fall-back to some supported formats (.mp3 ahem ahem) because OpenFL (SoundJS) would refuse to load the game (a known bug that was fixed but not integrated into OpenFL). That is the actual problem (also, **you should delete the default Ariel.ttf font for now if you want your game to load on Firefox and IE**, until OpenFL issue a fix.  

Bottom line, if you have to use an .ogg file with your html5 game (with SoundJS as an audio backend, but you can try Howler.js instead btw), what's working for all browsers (well, except for Opera Mini) is to use a .m4a audio file as a fall-back.

**Comment by MSGhero** [comment]
I just got hit by maybe this. HTML5 release on itch.io on Edge browser. The exact same conditions but on newgrounds.com works fine, as well as itch.io on Chrome.

For me, it was `beep.mp3` that caused the issue, but only `beep.ogg` is included in `bin.zip`. Very weird.

![unknown](https://user-images.githubusercontent.com/5033927/30516479-2aa0c90e-9b06-11e7-9553-8662c98f4727.png)

I'm also going to ask itch.io since it's only happening there.

**Comment by Gama11** [comment]
This is most likely no longer an issue with OpenFL 8 (although I have no test to test with Safari myself). If it still is, please let us know.

**Comment by nykevinwong** [comment]
hi Team,
my game in html5 build has a similar issue where the loading progress bar always stuck on iphone when it's close to finish loading. it happened to many

my game in html5 build: https://gaia-island.herokuapp.com
The same page always works fine for android browser as well as PC browser.
sorry that i have no IOS phone to get debug log about this.

**Comment by Gama11** [comment]
@nykevinwong We've just released 4.4.0, which along with OpenFL 8 has massively improved HTML5 output. Could you try with that? :)

https://haxeflixel.com/blog/12-HaxeFlixel-4-4-0/
